sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/UserAgentOptions.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.http.pipeline;
+
+import io.clientcore.core.annotations.Metadata;
+import io.clientcore.core.annotations.MetadataProperties;
+
+/**
+ * The {@code UserAgentOptions} class is used to configure the user agent string for HTTP requests.
+ */
+@Metadata(properties = MetadataProperties.FLUENT)
+public final class UserAgentOptions {
```

Comment:
Don't we need to add something to `HttpTrait` to consume this? Right now, there isn't a way for a service client to be built, unless we will be generating methods taking this directly rather than implementing an interface method, with a custom User-Agent policy.

Also, do we want to encapsulate full `UserAgentPolicy` configuration options here where a fully formed User-Agent could be set? This only allows configuring specific SDK and application ID information, making fully managed User-Agents more difficult.

Reply:
If I recall correctly we discussed it offline and decided that we can add convenience for reading `application.id` from configuration later. LMK if I remember it wrong